sdks/go/pkg/beam/core/runtime/xlangx/expansionx/download.go

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package expansionx contains utilities for starting expansion services for
+// cross-language transforms.
+package expansionx
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+)
+
+// URL is a type used to differentiate between strings and URLs
+type URL string
+
+const (
+	apacheRepository = URL("https://repo.maven.apache.org/maven2")
+	beamGroupID      = "org/apache/beam"
+	jarCache         = "~/.apache_beam/cache/jars"
+)
```

Comment:
Nit: In Java or Python, it's dictated that consts should be always at the top of the file.

In Go, it's preferable to define these closer to where they're used, since they are not hard to find anyway. Consider moving the const block to just above the`getURLForBeamJar` function.

Reply:
Moved for clarity